app/src/renderer/App.tsx

```diff
@@ -1,65 +1,30 @@
-import { Button } from "antd";
+import { Routes, Route, useNavigate } from "react-router";
+import { useEffect } from "react";
+
+import InboxView from "./views/Inbox";
+import SignInView from "./views/SignIn";
+import { useAppDispatch, useAppSelector } from "./hooks";
+import { clear } from "./features/session/sessionSlice";
 
 function App() {
-  const dummyRequest = async function () {
-    console.log("sending dummy request");
-    const res = await window.electronAPI.request({
-      method: "GET",
-      path_query: "/json",
-      headers: {},
-      stream: false,
-    });
-    console.log("received dummy response");
-    console.log(res);
-  };
+  const session = useAppSelector((state) => state.session);
+  const dispatch = useAppDispatch();
+  const navigate = useNavigate();
 
-  const dummyStreamRequest = async function () {
-    console.log("sending dummy stream request");
-    const res = await window.electronAPI.requestStream(
-      {
-        method: "GET",
-        path_query: "/html",
-        headers: {},
-        stream: true,
-      },
-      "/tmp/download",
-    );
-    console.log("received dummy stream response");
-    console.log(res);
-  };
+  // If we're not in offline mode and there's no session, redirect to sign-in
+  useEffect(() => {
```

Comment:
this seems good for now, but as we add more pages + components we may want to move towards using authentication context + protected routes to avoid exposing views to unauthed users. from doing some initial scoping, the approach outlined here seems pretty standard: https://www.robinwieruch.de/react-router-authentication/ 

wdyt?